rpcclient/local.go

```diff
@@ -180,6 +190,13 @@ func (l *Local) BroadcastTxSync(ctx context.Context, tx types.Tx) (*ctypes.Resul
 	}
 	res := <-resCh
 	r := res.GetCheckTx()
+
+	// broadcast tx
+	err = l.node.P2P.GossipTx(ctx, tx)
+	if err != nil {
+		return nil, fmt.Errorf("tx added to local mempool but failed to gossip: %w", err)
+	}
```

Comment:
Wasn't 100% sure what to do in this situation. The tx could still be included in a block but an error will be returned as it's not gossiped.

wdyt?

Reply:
Agree with Tomasz but what was the previous behaviour here (and what happens in tendermint in this case btw)?

Reply:
I think it is wrong to remove a valid Tx from the mempool only bc gossiping failed. What if there is a temporary networking error?
https://github.com/celestiaorg/optimint/blob/4097620751907e1bf0f1a4b5839d34b7ffcd5680/rpcclient/local.go#L194-L197

Reply:
It's really hard to see that the rest of the code is correct IMO.
Also, why does the rpcclient do the broadcasting here?

Reply:
Oh, I think I see: https://github.com/celestiaorg/optimint/pull/97#discussion_r708244572
We should document this better IMO.

Reply:
> what happens in tendermint in this case btw)?

afaict, this scenario doesn't occur in tendermint, as the tx is gossiped as part of the mempool reactor, so as long as the tx is in the mempool, it is being gossiped. 

This is not the case for here. If the user submits a tx and if fails to be gossiped, then it will remain in the local mempool and will never be gossiped. In fact, unless we remove it from the mempool, the user will be unable to try and gossip that transaction again using the same node, because when they do `CheckTx` will return an error indicating that the tx already exists in the mempool.